### PR TITLE
add UI for certifying organisation create and details page

### LINKED ISSUE #398: Create register certifying organisation view
#397 

create a register view to register a certifying organisation, including dropdown menu and details.

### PR DESCRIPTION
**Pull request by ann26**
fix #398 

*certification menu*
screenshot from 2017-04-17 20-18-58

*certifying organisation create* **update**
screenshot from 2017-05-05 17-59-13
screenshot from 2017-05-05 18-13-45

*no certifying organisation are defined*
screenshot from 2017-04-18 13-38-25

*certifying organisation lists - this page show up is when a user hit organisation button from certification menu*
*view for organisation owner and staff user*
screenshot from 2017-04-19 13-29-11
*view for other user*
screenshot from 2017-04-20 15-58-58

*certifying organisation details* 
*view for organisation owners and staff users*
screenshot from 2017-04-20 16-02-08
screenshot from 2017-04-20 16-02-20
screenshot from 2017-04-20 16-02-51
screenshot from 2017-04-20 16-03-02

*view for user who is a course convener - they can only edit course*
screenshot from 2017-04-21 14-17-12
screenshot from 2017-04-21 14-11-26
screenshot from 2017-04-21 14-16-02
screenshot from 2017-04-21 14-16-14

*view for other users*
screenshot from 2017-04-20 15-59-51
screenshot from 2017-04-20 16-00-05
screenshot from 2017-04-20 16-00-18
screenshot from 2017-04-20 16-00-29

*when no training center, course types, course, course convener are defined*
screenshot from 2017-04-21 14-44-51
screenshot from 2017-04-21 14-45-12
screenshot from 2017-04-21 14-45-32

*delete view*
screenshot from 2017-04-19 16-43-05

### CODE REVIEW

**Comment by timlinux** [comment]
@geomenke FYI

**Comment by timlinux** [comment]
Looking good @ann26 here are some notes:

screen shot 2017-04-21 at 1 25 19 pm

Can you please use a better widget for selecting owners so that you can type and the name list gets filtered to show matches. Also they should be listed alphabetically.

**Comment by timlinux** [comment]
screen shot 2017-04-21 at 1 33 05 pm

Clicking on the buttons here does not do anything for me.

**Comment by ann26** [comment]
@timlinux thanks, I'll look into it.
for those buttons, they are not working here, they will be made in separate pull requests

**Comment by meomancer** [comment]
Note :
this PR scope just

- [ ] Create certifying organization
- [ ] Accept/reject pending certifying organization
- [ ] Display detail

My note maybe:
1. Make validation for email, now we could input non-email format
2. Found bug on organisation (i'm not sure with others model)
So basically, we can create same name of organization. It will gives same slug both of them, and when we open the detail of organization or maybe delete one of organization, it will error, returned 2. 
how to produce:
- create 2 cert organization with same name
- go to one of organization

Solution maybe :
- Make name of organization uniquer
or
- Update save method, create method to generate slug, for example make slug 'organization-1', 'organization-2' if there are cert org that already named 'organization'

3. Not sure what map tells about in detail. It is location of organization? 
- Maybe we need to call google map api to get location based on address and country, and we safe the point that returned from google.
or
-Maybe no need map location of organization?

what do you think @timlinux ? for number 2 & 3

**Comment by ann26** [comment]
thanks @meomancer 
I updated the code and added the email validators
screenshot from 2017-05-02 13-37-15

File: django_project/certification/models/certifying_organisation.py [hunk 1]
```diff
@@ -59,6 +62,16 @@ def get_queryset(self):
         ).filter(approved=False)
 
 
+def validate_email_address(value):
+    try:
+        validate_email(value)
+        return True
+    except ValidationError(
+            _('%(value)s is not proper gitter room name'),
```
Comment (meomancer) [inline django_project/certification/models/certifying_organisation.py]: could you change the value error @ann26 ?
is not a valid email?

Comment (ann26) [inline django_project/certification/models/certifying_organisation.py]: Ah yes sure

**Comment by timlinux** [comment]
@meomancer 

For 2: I like the scheme of organisation-1, organisation-2 for the slug but maybe just validating as unique is enough because it is extremely unlikely that we have two companies with the same name.

For 3: Location shows where the training centers are - so eventually we can make a map of all training centers in the world. I think we don't need to map the organization itself other than to the country level. I'm ok with just letting the user click the position on the map for now in the training center form - its a very low frequency activity. If you have the geocoding stuff from feet easily able to be cut & pasted into the project codebase we can use that but otherwise lets just let the user do it manually by clicking the place on the map.

**Comment by ann26** [comment]
Hi @timlinux , so travis has made updates after may 4th (https://blog.travis-ci.com/2017-05-04-precise-image-updates) , 
which makes my recent commits failed in travis check. I did some experiment with the travis yml file in #411 and it turns out I had to add `sudo: false` in the file. However, I am not sure if this change is good. What do you think?

**Comment by lucernae** [comment]
I don't know if this will help:
https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments

It might be because previously, the default for this repository (when sudo is not explicitly stated in .travis.yml) is sudo: false. But since 4th May star wars day, they default to sudo: required? then causing it to use the latest edge precise build.

**Comment by timlinux** [comment]
@ann26 fine to change sudo if needed as long as the tests still function in the same way....

**Comment by ann26** [comment]
@timlinux ok

File: django_project/certification/migrations/0004_auto_20170505_1230.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    dependencies = [
+        ('certification', '0003_auto_20170505_1132'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='certifyingorganisation',
+            name='address',
+            field=models.TextField(help_text=b'Address of Organisation or Institution.', max_length=1000),
```
Comment (meomancer) [inline django_project/certification/migrations/0004_auto_20170505_1230.py]: Use ugettext_lazy for translation, but i think we could make another tickets.

**Comment by meomancer** [comment]
LGTM, for me

**Pull request merged**